admin-front/src/components/TimeSeries/functions/getTimeSeriesFilteredByOccurrence.ts

```diff
@@ -0,0 +1,53 @@
+import moment from 'moment/moment';
+import { Occurrence, TimeSeriesRecord } from '../utility';
+
+interface GetTimeSeriesFiltered {
+  timeSeries: TimeSeriesRecord[];
+  occurrence: Occurrence;
+}
+
+const MonthToIdMap = {
+  All: 0,
+  January: 1,
+  February: 2,
+  March: 3,
+  April: 4,
+  May: 5,
+  June: 6,
+  July: 7,
+  August: 8,
+  September: 9,
+  October: 10,
+  November: 11,
+  December: 12,
+};
+
+const getMonthIdFromTimeStamp = (timestamp: number): number => {
+  return Number(moment.unix(timestamp).format('M'));
+};
+
+export const getYearFromTimeStamp = (timestamp: number | undefined) => {
+  return Number(moment.unix(timestamp ?? 0).format('YYYY'));
+};
+
+export const getTimeSeriesFilteredByOccurrence = ({
+  timeSeries,
+  occurrence,
+}: GetTimeSeriesFiltered): TimeSeriesRecord[] => {
+  const newTimeSeries = [...timeSeries];
+
+  let timeSeriesFilteredByYear = newTimeSeries.filter((record) => {
```

Comment:
Personally not a fan of variable names this long (it's too noisy, not enough signal per characters). I know this is very subjective though. 

But, why not name this `filteredByYear`? Because we know from context that this bit of code is about time series (function name starts with `getTimeSeries..`).

Taking this further it can even be further reduced to just `byYear`? Because again, from context we know that this bit of code is about getting time series filtered by something.